src/main/RemoteRequest.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * RemoteRequest is used to for http requests on a remote server which can be
+ * fetched in chunks, e.g. using a Range request.
+ * @flow
+ */
+'use strict';
+
+import Q from 'q';
+
+type Chunk = {
+  start: number;
+  stop: number;
+  buffer: Object; // TODO: more efficient data store
```

Comment:
Exactly. We talked on Monday with you folks about replacing GA4GHDataSource with a generic access point. If you think it would be better in the folder sources I can move it there. But generally I was thinking we leave RemoteRequest in the main folder like RemoteFile and make specific datasources (like GA4GHDataSource, CoverageDataSource, etc.) in the sources folder which use RemoteRequest.js.  See https://github.com/akmorrow13/pileup.js/tree/master/src/main/sources <https://github.com/akmorrow13/pileup.js/tree/master/src/main/sources>.

As far as caching goes, RemoteFile has a generic cache as well that keeps Data sources from accessing the network. GA4GHDatatSource does not have a cache so we could swap RemoteRequest in here for the cache.

Best,

Alyssa Morrow
akmorrow@berkeley.edu <mailto:akmorrow@berkeley.edu>
414-254-6645






> On Jul 12, 2016, at 12:54 PM, Dan Vanderkam <notifications@github.com> wrote:
> 
> like a source class than a data class.